dev/tests/unit/Magento/FunctionalTestFramework/Util/Path/FilePathFormatterTest.php

```diff
@@ -33,52 +36,54 @@ public function testFormat($path, $withTrailingSeparator, $expectedPath)
     }
 
     /**
-     * Test file format with exception
+     * Test file format with exception.
      *
-     * @dataProvider formatExceptionDataProvider
      * @param string $path
-     * @param boolean $withTrailingSeparator
+     * @param bool $withTrailingSeparator
+     *
      * @return void
      * @throws TestFrameworkException
+     * @dataProvider formatExceptionDataProvider
      */
-    public function testFormatWithException($path, $withTrailingSeparator)
+    public function testFormatWithException(string $path, bool $withTrailingSeparator): void
     {
         $this->expectException(TestFrameworkException::class);
         $this->expectExceptionMessage("Invalid or non-existing file: $path\n");
         FilePathFormatter::format($path, $withTrailingSeparator);
     }
 
     /**
-     * Data input
+     * Data input.
      *
      * @return array
      */
-    public function formatDataProvider()
+    public function formatDataProvider(): array
     {
         $path1 = rtrim(TESTS_BP, '/');
         $path2 = $path1 . DIRECTORY_SEPARATOR;
+
         return [
-            [$path1, null, $path1],
+            [$path1, false, $path1],
```

Comment:
@bohdan-harniuk 
I see you have changed `null` to `false` in many places for data input. Isn't it the opposite of the original test. Am I missing something else?

Reply:
Hello, @jilu1!

I've done it after the small investigation. I believe that bool variables should have only two states: TRUE or FALSE.

There are 55 places, in the MFTF source code, where `FilePathFormatter::format` method is called.
Also, there are 17 places in the MFTF source code where `UrlFormatter::format` method is called.

**I've checked all those places, if in any of them there is null value passed into those methods. There is no such place.**
So, I decided to limit the value of this variable: TRUE or FALSE.
**The NULL values was passed only during testing for both cases.**

Do we have some other intentions to allow pass here NULL value (?bool instead of bool)?

Thanks, Bohdan

Reply:
@bohdan-harniuk 
Maybe we can use `?bool` here. The intention is when `NULL` is used, the relevant parameter is omitted and thus to test using the default value in `FilePathFormatter::format` or `UrlFormatter::format`

Reply:
@jilu1 
It makes sense! I'll rollback NULL values.
Thank you!